src/test/java/org/sonar/plugins/checkstyle/CheckstyleRulesDefinitionTest.java

```diff
@@ -61,7 +61,7 @@ public void test() {
         assertThat(repository.language()).isEqualTo("java");
 
         final List<RulesDefinition.Rule> rules = repository.rules();
-        assertThat(rules).hasSize(179);
+        assertThat(rules).hasSize(353);
```

Comment:
this should be 179 * 2 = 358 - (amount of Checks that are new)

Looks like we have 5 new Checks, but I see only 4 new checks https://checkstyle.org/releasenotes.html 

IllegalIdentifierName
RecordComponentNumberCheck
RecordTypeParameterNameCheck
PatternVariableName.

https://checkstyle.org/releasenotes.html#Release_8.33 we introduced NoCodeInFile and it is old metadata, so most likely has to be duplicated.

Please help to explain why `353`.

Reply:
I found missed by me Check - https://checkstyle.org/config_javadoc.html#RequireEmptyLineBeforeBlockTagGroup
there was mistake in release notes, bad issue title.

fixed.

Reply:
I updated code to validate counts more clearly